src/library_html5.js

```diff
@@ -988,13 +999,14 @@ var LibraryHTML5 = {
   emscripten_set_orientationchange_callback_on_thread__proxy: 'sync',
   emscripten_set_orientationchange_callback_on_thread__deps: ['$registerOrientationChangeEventCallback'],
   emscripten_set_orientationchange_callback_on_thread: (userData, useCapture, callbackfunc, targetThread) => {
-    if (!screen || !screen['addEventListener']) return {{{ cDefs.EMSCRIPTEN_RESULT_NOT_SUPPORTED }}};
-    return registerOrientationChangeEventCallback(screen, userData, useCapture, callbackfunc, {{{ cDefs.EMSCRIPTEN_EVENT_ORIENTATIONCHANGE }}}, "orientationchange", targetThread);
+    if (!screen || !screen.orientation) return {{{ cDefs.EMSCRIPTEN_RESULT_NOT_SUPPORTED }}};
```

Comment:
Good point in your previous comment: the check `if (!screen)` here is incorrect, as it would throw an exception if `screen` doesn't exist. `if (!window.screen)` would work as a replacement.

Reply:
Best is probably to get the support check in a single place, like other code does, e.g.:

https://github.com/emscripten-core/emscripten/blob/f7e28e58a81e1ae7cd6b73c23045bc4c8a3ebd8e/src/library_html5.js#L1010

I'd prefer to land the rest of this then go back to change that (since it's been this way for a while and needs some time with older OSes and browser versions to verify the correct way).